src/librustdoc/test.rs

```diff
@@ -241,7 +241,12 @@ fn run_test(
     };
     let output_file = outdir.path().join("rust_out");
 
-    let mut compiler = Command::new(std::env::current_exe().unwrap().with_file_name("rustc"));
+    let mut compiler = if let Some(rustc) = env::var_os("RUSTC"){
+        Command::new(&rustc)
+    }else {
+        let rustc = std::env::current_exe().unwrap_or_default().with_file_name("rustc");
```

Comment:
Let's not use default here but instead have a separate else branch where we do Command::new("rustc").

Reply:
sure have fixed it now but do not feel that I need to waste more ci time with pushing it until I have a path forward with the failing tests